server/channels/api4/hosted_customer.go

```diff
@@ -4,402 +4,19 @@
-		return
-	}
-
-	req.ServerID = c.App.Srv().TelemetryId()
-
-	if err := c.App.Cloud().SubscribeToNewsletter("", req); err != nil {
-		c.Err = model.NewAppError(where, "api.server.cws.subscribe_to_newsletter.app_error", nil, "CWS Server failed to subscribe to newsletter.", http.StatusInternalServerError).Wrap(err)
-		return
-	}
-
-	ReturnStatusOK(w)
-}
-
-func selfHostedConfirmExpand(c *Context, w http.ResponseWriter, r *http.Request) {
-	const where = "Api4.selfHostedConfirmExpand"
-
-	ensureSelfHostedAdmin(c, where)
-	if c.Err != nil {
-		return
-	}
-
-	if !checkSelfHostedPurchaseEnabled(c) {
-		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusNotImplemented)
-		return
-	}
-
-	bodyBytes, err := io.ReadAll(r.Body)
-	if err != nil {
-		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusBadRequest).Wrap(err)
-		return
-	}
-
-	var confirm model.SelfHostedConfirmPaymentMethodRequest
-	err = json.Unmarshal(bodyBytes, &confirm)
-	if err != nil {
-		c.Err = model.NewAppError(where, "api.cloud.request_error", nil, "", http.StatusBadRequest).Wrap(err)
-		return
-	}
-
-	user, userErr := c.App.GetUser(c.AppContext.Session().UserId)
-	if userErr != nil {
-		c.Err = userErr
-		return
-	}
-
-	confirmResponse, err := c.App.Cloud().ConfirmSelfHostedExpansion(confirm, user.Email)
-	if err != nil {
-		if confirmResponse != nil {
-			c.App.NotifySelfHostedSignupProgress(confirmResponse.Progress, user.Id)
-		}
-
-		if err.Error() == strconv.Itoa(http.StatusUnprocessableEntity) {
-			c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusUnprocessableEntity).Wrap(err)
-			return
-		}
-		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-		return
-	}
-
-	license, appErr := c.App.Srv().Platform().SaveLicense([]byte(confirmResponse.License))
-	// dealing with an AppError
-	if appErr != nil {
-		if confirmResponse != nil {
-			c.App.NotifySelfHostedSignupProgress(confirmResponse.Progress, user.Id)
-		}
-		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-		return
-	}
-	clientResponse, err := json.Marshal(model.SelfHostedSignupConfirmClientResponse{
-		License:  utils.GetClientLicense(license),
-		Progress: confirmResponse.Progress,
-	})
-	if err != nil {
-		if confirmResponse != nil {
-			c.App.NotifySelfHostedSignupProgress(confirmResponse.Progress, user.Id)
-		}
-		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-		return
-	}
-
-	go func() {
-		err := c.App.Cloud().ConfirmSelfHostedSignupLicenseApplication()
-		if err != nil {
-			c.Logger.Warn("Unable to confirm license application", mlog.Err(err))
-		}
-	}()
-
-	_, _ = w.Write(clientResponse)
+	c.Err = model.NewAppError(where, "api.server.hosted_signup_unavailable.error", nil, "", http.StatusNotImplemented)
```

Comment:
Also, we may need to remove all webapp client functions to all these endpoints that are being removed

Reply:
This is mostly left around in case I missed somewhere in the webapp. This will ensure that the check returns an expected error rather than a 404/500